src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -10,42 +15,191 @@ namespace Infrastructure.Common
+                // This convention permits a developer to override any detection logic to
+                // enable or disable manually.
+                string environmentVariable = Environment.GetEnvironmentVariable(conditionName);
+                bool result = false;
+                if (!String.IsNullOrWhiteSpace(environmentVariable) && bool.TryParse(environmentVariable, out result))
+                {
+                    return result;
+                }
+                
+                // No enviromnent variable, so detect at runtime or default to 'false'
+                return detectFunc == null ? false : detectFunc();
+            });
+        }
 
-            return false;
+        // Returns the 'true' or 'false' for a condition whose name is
+        // contained in TestProperties.   If the property's value is
+        // 'true' or 'false', it will be used as is.  If the value is
+        // empty, and 'detectFunc' is specified, it will be invoked to
+        // evaluate the condition at runtime.  The value is cached, so
+        // 'detectFunc' will be called only once.
+        private static bool GetConditionValueFromTestProperties(string conditionName, Func<bool> detectFunc = null)
+        {
+            string propertyValue;
+            return GetConditionValue(conditionName, () =>
+            {
+                propertyValue = TestProperties.GetProperty(conditionName);
+                if (string.IsNullOrWhiteSpace(propertyValue))
+                {
+                    // Empty string means auto-detect if possible, else false
+                    return (detectFunc == null) ? false : detectFunc();
+                }
+                else
+                {
+                    bool conditionResult = false;
+                    if (!bool.TryParse(propertyValue, out conditionResult))
+                    {
+                        conditionResult = false;
+                    }
+
+                    return conditionResult;
+                }
+            });
+        }
+
+        private static bool Detect_Server_Is_Local()
+        {
+            // Temporary workaround to detect whether this test is using localhost
+            // for either new or old test infrastructure.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            string host = null;
+            if (TestProperties.PropertyNames.Contains("ServiceUri"))
+            {
+                host = TestProperties.GetProperty("ServiceUri");
+            }
+            else if (TestProperties.PropertyNames.Contains("BridgeHost"))
+            {
+                host = TestProperties.GetProperty("BridgeHost");
+            }
+             if (String.IsNullOrWhiteSpace(host))
+            {
+                return false;
+            }
+            int index = host.IndexOf("localhost", 0, StringComparison.OrdinalIgnoreCase);
+            return index >= 0;
+        }
```

Comment:
I think we need an unambiguous test infrastructure method we can call for this.  The logic doesn't belong here.  Bear in mind, part of this initial checkin is bounded by making some simple heuristic decisions for logic we probably want supported by the test infrastructure.